plugins/store/guest-js/index.ts

```diff
@@ -8,50 +8,225 @@ import { invoke, Resource } from '@tauri-apps/api/core'
 
 interface ChangePayload<T> {
   path: string
+  resourceId?: number
   key: string
-  value: T | null
+  value: T
+  exists: boolean
 }
 
 /**
  * Options to create a store
  */
 export type StoreOptions = {
   /**
-   * Auto save on modification with debounce duration in milliseconds
+   * Auto save on modification with debounce duration in milliseconds, it's 100ms by default, pass in `false` to disable it
    */
-  autoSave?: boolean
+  autoSave?: boolean | number
+  /**
+   * Name of a serialize function registered in the rust side plugin builder
+   */
+  serializeFnName?: string
+  /**
+   * Name of a deserialize function registered in the rust side plugin builder
+   */
+  deserializeFnName?: string
+}
+
+/**
+ * @param path: Path to save the store in `app_data_dir`
+ * @param options: Store configuration options
+ *
+ * @throws If a store at that path already exists
+ */
+export async function createStore(
+  path: string,
+  options?: StoreOptions
+): Promise<Store> {
+  return await Store.createStore(path, options)
 }
 
 /**
+ * Create a new Store or get the existing store with the path
+ *
  * @param path: Path to save the store in `app_data_dir`
  * @param options: Store configuration options
  */
-export async function createStore(path: string, options?: StoreOptions) {
-  const resourceId = await invoke<number>('plugin:store|create_store', {
-    path,
-    ...options
-  })
-  return new Store(resourceId, path)
+export async function createOrExistingStore(
+  path: string,
+  options?: StoreOptions
+): Promise<Store> {
+  return await Store.createOrExistingStore(path, options)
+}
+
+/**
+ * @param path: Path of the store in the rust side
+ */
+export async function getStore(path: string): Promise<Store | undefined> {
+  return await Store.getStore(path)
 }
 
 /**
  * A lazy loaded key-value store persisted by the backend layer.
  */
-export class Store extends Resource {
+export class LazyStore implements IStore {
```

Comment:
I would also consider this a breaking change as well and we can't just change the behavior of normal `Store` and add a new `LazyStore`type.

I also honestly don't see why we need two Store types, I believe it can just be one.

Reply:
The normal `Store`'s behavior was changed in the last rework PR, this PR didn't change anything about it, just added a new one

Reply:
I see, so it is just outdated/incorrect documentation then. I am fine with keeping `LazyStore` but I don't find a really good use for it? I think `Store` should just be initialized once the class is constructed and so users have full control over the open/read/write/close

Reply:
Actually I would want to promote `LazyStore` a bit more since you don't need to handle the async creation of the store and also you get the lazy load

Reply:
Let's keep it, although I wouldn't want it to be the default in README and docs